iOS/Flows/Conversation/Cells/ConversationMessageCellLayout.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  ConversationMessageCellLayout.swift
+//  Jibber
+//
+//  Created by Martin Young on 11/4/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+
+class ConversationMessageCellLayout: UICollectionViewFlowLayout {
+
+    override class var layoutAttributesClass: AnyClass {
+        return ConversationMessageCellLayoutAttributes.self
+    }
+
+    override func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        guard let attributesInRect = super.layoutAttributesForElements(in: rect) else { return nil }
+
+        for attributes in attributesInRect {
+            guard let messageAttributes = attributes as? ConversationMessageCellLayoutAttributes else {
+                continue
+            }
+            self.update(attributes: messageAttributes)
+        }
+
+        return attributesInRect
+    }
+
+    override func layoutAttributesForItem(at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        let attributes = super.layoutAttributesForItem(at: indexPath)
+
+        if let messageAttributes = attributes as? ConversationMessageCellLayoutAttributes {
+            self.update(attributes: messageAttributes)
+        }
+
+        return attributes
+    }
```

Comment:
I take the attributes returned from the flow layout as a starting point and modify them as needed.